### PR TITLE
[JENKINS-54325] Add support for multiple footers in SimplePageDecorator

### PR DESCRIPTION
**Pull request by jvz**
Signed-off-by: Matt Sicker <boards@gmail.com>

See [JENKINS-54325](https://issues.jenkins-ci.org/browse/JENKINS-54325). This supersedes #3710.

### Proposed changelog entries

* `Internal:` `login.jelly` and `signup.jelly` now support displaying all registered `SimplePageDecorator` extensions in the footer rather than just the top priority one.

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@reviewbybees @scherler @Wadeck @jeffret-b @jenkinsci/code-reviewers

### CODE REVIEW

**Comment by jvz** [comment]
Rebuilding PR

File: core/src/test/java/hudson/UtilTest.java [hunk 1]
```diff
@@ -352,6 +355,7 @@ public void testDeleteContentsRecursive() throws Exception {
     }
 
     @Test
+    @Retry
```
Comment (jglick) [inline core/src/test/java/hudson/UtilTest.java]: This just duplicates the Surefire feature which is already enabled in the repo.

If the test is flaky enough that it is failing for you five times in a row, just use

```java
@Ignore("TODO fails frequently in CI")
```

and leave it for a rainy day for someone to figure out why it was failing and how to fix it, or to decide it is worthless and just delete it.

Comment (jvz) [reply core/src/test/java/hudson/UtilTest.java]: The surefire feature is a global setting, while this one is localized.

**Review by scherler — APPROVED** [review]
LGTME thank you very much!

**Review by Wadeck — APPROVED** [review]
:bee:

**Review by jeffret-b — APPROVED** [review]
Looks good.

**Comment by Wadeck** [comment]
Re-run the test, due to "Caused by: java.net.BindException: Address already in use"

**Comment by Wadeck** [comment]
If the test passed this time and no negative feedback, we will merge it into the weekly tomorrow.

**Comment by Wadeck** [comment]
@jvz Seems that `testDeleteRecursive_onWindows` should be ignored or corrected like the other :(

**Review by Wadeck — APPROVED** [review]
Both tests that are failing:
- hudson.util.AtomicFileWriterPerfTest
- hudson.FilePathSEC904Test

Are known and in investigation state.

**Comment by batmat** [comment]
Last PR build is:
image

Which is about to be ignored in #3797 

So we can consider the current PR as "green" :).

File: core/src/main/resources/jenkins/model/Jenkins/login.jelly [hunk 2]
```diff
@@ -50,6 +50,7 @@ THE SOFTWARE.
   <!-- in case of error we want to surround the form elements with an error hint -->
   <j:set var="inputClass" value="${error ? 'danger' : 'normal'}"/>
   <j:set var="simpleDecorator" value="${h.simplePageDecorator}"/>
```
Comment (batmat) [inline core/src/main/resources/jenkins/model/Jenkins/login.jelly]: It looks to me this should be deleted, given AFAIU now `simpleDecorator` is not used anymore ?

File: core/src/main/java/hudson/Functions.java [hunk 3]
```diff
@@ -1779,6 +1779,10 @@ public static SimplePageDecorator getSimplePageDecorator() {
         return SimplePageDecorator.first();
```
Comment (batmat) [inline core/src/main/java/hudson/Functions.java]: I wonder if this makes much sense. Should we deprecate this?

**Review by batmat — CHANGES_REQUESTED** [review]
LGTM, but I think we must remove the now unused `simpleDecorator` in jelly (and likely deprecate the `getSimplePageDecorator` method to avoid raising the WTF/line ratio).

**Comment by Wadeck** [comment]
@batmat see the initial discussion on those points in #3710 which is mentioned in the description.
TL;DR: it's important to keep them

**Comment by jvz** [comment]
@batmat I already tried that approach first and was rejected.

**Review by batmat — APPROVED** [review]
:+1: then.

**Comment by batmat** [comment]
I think enough approval were given here. Let's merge this some time tomorrow or later today if nobody objects in the meantime :).

\o/
